### ISSUE TITLE
WebView2 sending typed form contents to nleditor.osi.office.net

### ISSUE THREAD

**Issue by RendijsSmukulis** [MicrosoftEdge/WebView2Feedback#2865]
**Description**

After updating runtime to `106.0.1370.37` we noticed that whenever user is typing into a form, the text is sent off to a Microsoft endpoint, the text is in the see the following json's Content.TileElements.Text field:

```
POST https://nleditor.osi.office.net/NLEditor/TileCheck/V1 HTTP/1.1
Host: nleditor.osi.office.net
Connection: keep-alive
Content-Length: 385
Content-Type: application/json
Sec-Fetch-Site: none
Sec-Fetch-Mode: no-cors
Sec-Fetch-Dest: empty
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/106.0.0.0 Safari/537.36 Edg/106.0.1370.37
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9

{"SessionId": "B567EF182BE92B9246C18756A8204F4B","AppId": "Edge_Win32","LanguageUxId": "en-US","Content": [{"TileId": "en-US","RevisionId": "0","TileElements": [{"LanguageId": "en-US","Text": "foo baz bar","TextUnit": 8}]}],"Descriptors":[{"Name": "FlightIds","Value": "wac-wordeditorservicemultiplegrammarcritiquespersentence-treatment"},{"Name": "LicenseType","Value": "NoLicense"}]}
```

This appears to be a spelling/grammar service. Sending the user-typed text is a significant privacy issue for us - is there a flag or some other way to disable this feature programmatically?

**Version**
SDK: 1.0.1222-prerelease
Runtime: 106.0.1370.37
Framework: WPF
OS: Win10

**Repro Steps**

- Start a MITM proxy to monitor the traffic (e.g. Fiddler)
- Navigate to a site with a form ( e.g. https://www.w3schools.com/html/html_form_input_types.asp )
- Type in some text (with spaces) in the form
- Observe the multiple requests sent off to https://nleditor.osi.office.net/NLEditor/TileCheck/V1

**Screenshots**

**Additional context**

[AB#41963725](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/41963725)

**Comment by zewt** [comment]
> Sending the user-typed text is a significant privacy issue for us

This would be a significant privacy issue for everybody.  Has anybody confirmed this?  I don't want to knee-jerk, but this is a critical bug if accurate.  It's too late for me to look at it right now, but I'll try to take a look tomorrow.

**Comment by victorthoang** [comment]
Hello @RendijsSmukulis,

Thanks for your bug report. I've assigned this to a dev that can best try to replicate the issue and follow up on this.

**Comment by ElyssaJyu** [comment]
Hi @RendijsSmukulis, let me confirm your concern that spell check service may have a privacy issue (correct me if not).

As previously reported in the issue https://github.com/MicrosoftEdge/WebView2Feedback/issues/2876, WebView2 will match the behavior of the browser, so WebView2 has this feature on by default.  You can disable by setting the extra browser arguments: `--disable-features=msUseSpellCheckCorrectionsCard`.
There is some related information about Microsoft Editor Service in Edge Browser which provide speck checking. 
[Spell checking provided by Microsoft Editor](https://learn.microsoft.com/en-us/DeployEdge/microsoft-edge-policies#microsofteditorproofingenabled); [Writing assistance](https://learn.microsoft.com/en-us/microsoft-edge/privacy-whitepaper/#writing-assistance)
> Microsoft Edge sends typed text and a service token to a Microsoft cloud service over a secure HTTPS connection. The service token doesn't contain any user-identifiable information. A Microsoft cloud service then processes the text to detect spelling and grammar errors in your text. All your typed text that's sent to Microsoft is deleted immediately after processing occurs. No data is stored for any period of time.

**Comment by zewt** [comment]
Any reasonable privacy policy will assure users that their data won't be sent to third parties without their consent, and this causes developers to violate their own privacy policy without any way of knowing short of auditing traffic in every patch (or stumbling on GitHub tickets like I did).  Developers need to be able to trust the platform their applications run on.

**Comment by ElyssaJyu** [comment]
Hi @zewt, We've added this as a bug on our backlog and track it internally.

**Comment by champnic** [comment]
This has been fixed in runtimes version 109.0.1481.0+. Thanks!

@zewt I'm forwarding your concern to our folks looking into potential privacy policy issues. I agree we need some testing around this to catch instances of other teams on the browser adding unexpected traffic.
